### PR TITLE
Workflow to tag releases, and generate release notes

### PR DESCRIPTION
**Pull request by woody-apple**
Added a script and a workflow to generate releases, with release notes. 

Note: There's a new overall spec version file added to the repo, that represents the highest spec number that this release can be used for.

### CODE REVIEW

File: scripts/tagging/tag_new_release.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#
+# Copyright (c) 2020 Project CHIP Authors
```
Comment (ksperling-apple) [inline scripts/tagging/tag_new_release.sh:2]:
```suggestion
# Copyright (c) 2023 Project CHIP Authors
```

**Review by ksperling-apple — APPROVED** [review]

File: scripts/tagging/tag_new_release.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#!/bin/bash
+
+CURRENT_SPEC_VERSION=$(cat SPECIFICATION_VERSION)
+
+CURRENT_RELEASE=$(gh release list --exclude-pre-releases | grep -Fi "$CURRENT_SPEC_VERSION" | awk '{print $1}' | head -n1)
```
Comment (bzbarsky-apple) [inline scripts/tagging/tag_new_release.sh:21]: What is this line aiming to do?  Some documentation would be good.

File: scripts/tagging/tag_new_release.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+else
+    SDK_RELEASE_REVISIONS="$(echo "$CURRENT_RELEASE" | cut -d'.' -f4)"
+fi
+
+if [ ! -z "$CURRENT_RELEASE" ]; then
+    CURRENT_RELEASE="v$CURRENT_SPEC_VERSION.$SDK_RELEASE_REVISIONS"
+    SDK_RELEASE_REVISIONS=$(($SDK_RELEASE_REVISIONS + 1))
+fi
+
+NEW_RELEASE_TAG="v$CURRENT_SPEC_VERSION.$SDK_RELEASE_REVISIONS"
```
Comment (bzbarsky-apple) [inline scripts/tagging/tag_new_release.sh:35]: It woudl be good to document what the goal of this whole chunk is.

File: scripts/tagging/tag_new_release.sh (hunk 1 above)
Comment (terafin) [reply scripts/tagging/tag_new_release.sh]: Done

**Pull request merged**